packages/commonwealth/server/routes/updateChainCustomDomain.ts

```diff
@@ -0,0 +1,39 @@
+import { Response, NextFunction } from 'express';
+import { TypedRequestBody } from '../types';
+import { DB } from '../database';
+
+type updateCustomDomainReq = {
+  secret: string;
+  chain_id: string;
+  custom_domain: string;
+};
+
+const updateChainCustomDomain = async (
+  models: DB,
+  req: TypedRequestBody<updateCustomDomainReq>,
+  res: Response,
```

Comment:
You should use `TypedResponse` here as well, along with `success`. For errors, throw `AppError` and `ServerError` rather than `next`.